### PR TITLE
[GCI 2019] Added user guide to StreamRadio

### PR DESCRIPTION
**Pull request by trungnt2910**
I've done the task of adding an user guide for StreamRadio and solved #1
I've also added a screenshot to README.md. 
Please review my guide and give me some suggestions if editing is needed.

### CODE REVIEW

File: docs/userguide.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+*	[Installation](#part_install)
+*	[First run](#part_first_run)
+*	[Adding a station](#part_add_station)
+*	[Managing stations](#part_manage)
+	*	[The Station Panel](#manage_panel)
+	*	[Adding a station using Shoutcast URL](#manage_add)
+	*	[Probing a station](#manage_check)
+	*	[Deleting a station](#manage_delete)
+
+##Installation <a name = "part_install"></a>
```
Comment (Begasus) [inline docs/userguide.md]: Add a space between ## Installation

Comment (trungnt2910) [reply docs/userguide.md]: Thanks. I've fixed that.

File: docs/userguide.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+*	[Managing stations](#part_manage)
+	*	[The Station Panel](#manage_panel)
+	*	[Adding a station using Shoutcast URL](#manage_add)
+	*	[Probing a station](#manage_check)
+	*	[Deleting a station](#manage_delete)
+
+## Installation <a name = "part_install"></a>
+StreamRadio is available on HaikuDepot on both 32-bit and 64-bit Haiku. To install it, open HaikuDepot, and search for the term *StreamRadio*. Select *install* to download the package to your computer.
+
+![Installation](/screenshots/00_Installation.png)
```
[the remark this answers is not in the document]
Comment (trungnt2910) [reply docs/userguide.md]: So I should remove the installation part completely?

[the remark this answers is not in the document]
Comment (humdingerb) [reply docs/userguide.md]: I'd say it's OK to keep the short section, just remove the screenshot. It takes too much space compared its usefulness... :)

[the remark this answers is not in the document]
Comment (trungnt2910) [reply docs/userguide.md]: OK.

**Comment by humdingerb** [comment]
After you took the "first launch" screenshot, can you run all PNG through "optipng -o 5 *.png" (available from HaikuDepot)? That makes them nice and small.

File: docs/userguide.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+You can also build StreamRadio from source to get a newer (but unstable) version of 
+StreamRadio. The source is available [here](https://github.com/HaikuArchives/Haiku-Radio).
+
+## First run <a name = "part_first_run"></a>
+
+![FirstRun](/screenshots/01_First_run_GUI.png)
```
[the remark this answers is not in the document]
Comment (trungnt2910) [reply docs/userguide.md]: Well,  I managed to do that by clicking and holding on the *Search* menu then pressing PrtScr.

File: docs/userguide.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+#### Search by
+This menu allows you to select your search criteria. Depending on the radio service you choose, you can search based on keywords, genre, and countries.
+
+#### Search for
+This can either be a menu or a text box, according to the option you chose for *Search by*. Type in your keyword, or select your preferred type here.
+
+After you've done with your choices, press ![SearchIcon](bnSearch.png) or hit <kbd>Enter</kbd> to start your search. The search may take up to a minute.
+
+![NewStation](/screenshots/03_Add_New_Stations_2.png)
```
[the remark this answers is not in the document]
Comment (trungnt2910) [reply docs/userguide.md]: Oh, there was some change of order at the time I edited the guide but I forgot to change the tags.

**Comment by trungnt2910** [comment]
I've resolved all the issues. Please review my pull request.

**Comment by humdingerb** [comment]
Alright! All that's missing is how to play/stop a station. I suggest another short section "Start/Stop streaming a station" after the "Adding a station" section.

**Comment by trungnt2910** [comment]
@humdingerb I've added that part in my latest commit.

File: docs/userguide.md [hunk 5]
```diff
@@ -53,11 +54,24 @@ The results will appear in the box below. Select your preferred station, then pr
...
+
+The status bar should then indicate the name of the station you're currently playing.
+
+![Play](screenshots/04_Play_a_station.png)
+
+Click on the corresponding square on another station to switch to that station.
+
+To stop streaming, click on the square again.
+
+> **Note**: Stopping a stream this way may cause crashes in the current version of StreamRadio.
```
Comment (humdingerb) [inline docs/userguide.md]: Starting and stopping works here. Mentioning possible bugs doesn't really fit into a user guide. I'd remove this note, but add that double-clicking a station works also to start/stop streaming. This is much easier that aiming for a tiny icon...

File: docs/userguide.md [hunk 6]
```diff
@@ -53,11 +54,24 @@ The results will appear in the box below. Select your preferred station, then pr
...
 
+## Start/Stop streaming a station <a name = "part_start"></a>
+To start streaming a station, simply click on the little square on the bottom left corner of the station's icon. 
+
+The status bar should then indicate the name of the station you're currently playing.
+
+![Play](screenshots/04_Play_a_station.png)
+
+Click on the corresponding square on another station to switch to that station.
+
```
Comment (humdingerb) [inline docs/userguide.md]: Remove this empty line and put a few spaces in front of the next line to create a line break.

File: docs/userguide.md (hunk 5 above)
Comment (trungnt2910) [reply docs/userguide.md]: Interesting, because I haven't noticed that while learning how to use the app myself.

File: docs/userguide.md (hunk 6 above)
Comment (trungnt2910) [reply docs/userguide.md]: OK, I've made the changes.

File: docs/userguide.md (hunk 5 above)
Comment (Begasus) [reply docs/userguide.md]: On x86 it works ok for me too, so don't think it would be needed also

File: docs/userguide.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+The results will appear in the box below. Select your preferred station, then press *Add* to add it to your list of stations. You can continue searching and adding stations like this.
+
+When you're finished, close *Find Stations* to return to the main window.
+
+## Start/Stop streaming a station <a name = "part_start"></a>
+To start streaming a station, simply click on the little square on the bottom left corner of the station's icon. The status bar should then indicate the name of the station you're currently playing.  
+![Play](screenshots/04_Play_a_station.png)  
+Click on the corresponding square on another station to switch to that station. To stop streaming, click on the square again.  
+
```
Comment (Begasus) [inline docs/userguide.md]: There is no square when the stream is running, maybe rephrase?
SR-running

Comment (trungnt2910) [reply docs/userguide.md]: There is in my PC, both the version from HaikuDepot and the version compiled from source.

Comment (trungnt2910) [reply docs/userguide.md]: image
Is this an error?
I've never seen the play button before in StreamRadio.

Comment (Begasus) [reply docs/userguide.md]: That's probably before the stream got started

Comment (Begasus) [reply docs/userguide.md]: Once the stream started you should get the play button ...

Comment (trungnt2910) [reply docs/userguide.md]: No, the stream has started for a while, I'm sure of that.
Anyway, it may be related to my system, so I've adjusted the guide to match with normal behavior.

**Comment by humdingerb** [comment]
LGTM, merge?

**Comment by Begasus** [comment]
> LGTM, merge?

+1 for me :)

**Comment by humdingerb** [comment]
Thanks!
